svgdigitizer/svgfigure.py

```diff
@@ -0,0 +1,742 @@
+r"""
+Reconstructed scientific plot with special units and/or time axis.
+
+A :class:`FigurePlot` wraps a plot in SVG format consisting of a curve, axis
+labels and (optionally) additional metadata provided as text fields in the SVG.
+
+"""
+# ********************************************************************
+#  This file is part of svgdigitizer.
+#
+#        Copyright (C) 2021-2023 Albert Engstfeld
+#        Copyright (C) 2021-2023 Julian Rüth
+#
+#  svgdigitizer is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  svgdigitizer is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with svgdigitizer. If not, see <https://www.gnu.org/licenses/>.
+# ********************************************************************
+import logging
+from functools import cached_property
+
+import astropy.units as u
+
+from svgdigitizer.exceptions import SVGAnnotationError
+
+logger = logging.getLogger("figureplot")
+
+
+class SVGFigure:
+    """
+    TODO:: Add description and docstring (see issue #177)
+    """
+
+    def __init__(self, svgplot, metadata=None):
+        self.svgplot = svgplot
+        self._metadata = metadata or {}
+
+    @staticmethod
+    def create_figure(measurement_type=None):
+        """
+        Creates plots for specific data such as cyclic voltammograms (CV).
+        TODO:: Add description and docstring (see issue #177)
+        """
+        from svgdigitizer.electrochemistry.cv import CV  # pylint: disable=cyclic-import
```

Comment:
pylint and black want to have this line in different ways.
* black as is
* isort:
```
from svgdigitizer.electrochemistry.cv import \
            CV  # pylint: disable=cyclic-import
```

Reply:
I guess we should pass `--line-length 256` to isort.